### PR TITLE
Add new LegendItem component and LineChartLegend

### PR DESCRIPTION
**Pull request by cbauerman**
## Specs

Add a new `LegendItem` component and create a legend for the `LineChart` component with it, as well as refactoring the existing `PieChartLegend` component.

## Testing 

**TBD**

### CODE REVIEW

**Comment by Tug** [comment]
Fixed the problem (of selecting the path globally instead of in the current chart).
I think it's ready for review now

File: client/components/line-chart/index.js [hunk 1]
```diff
@@ -52,12 +53,14 @@ const dateFormatFunction = displayMonthTicksOnly => ( date, index, tickRefs ) =>
 };
 
 const dateToAbsoluteMonth = date => date.getYear() * 12 + date.getMonth();
+const NUM_SERIES = 3;
```
Comment (yurynix) [inline client/components/line-chart/index.js]: Can we have a comment or a more descriptive name to that? I'm not sure what it does
It's the number of lines on the graph?

File: client/components/line-chart/style.scss [hunk 2]
```diff
@@ -92,3 +123,42 @@
...
+	flex-wrap: wrap;
+	justify-content: flex-start;
+	cursor: pointer;
+	margin-right: 20px;
+
+	.legend-item {
+		margin-left: 20px;
+	}
+
+	@include breakpoint( )
```
Comment (yurynix) [inline client/components/line-chart/style.scss]: Are we missing here something?

File: client/my-sites/google-my-business/stats/chart.js [hunk 3]
```diff
@@ -55,6 +55,19 @@ function transformData( props ) {
 	} );
 }
 
+function createLegendInfo( props ) {
```
Comment (yurynix) [inline client/my-sites/google-my-business/stats/chart.js]: if we only use `data`, maybe just pass that?

File: client/components/line-chart/style.scss (hunk 2 above)
Comment (Tug) [reply client/components/line-chart/style.scss]: Right, we don't need that anymore now that the legend items wrap

File: client/my-sites/google-my-business/stats/chart.js (hunk 3 above)
Comment (Tug) [reply client/my-sites/google-my-business/stats/chart.js]: We use `dataSeriesInfo` as well here

**Pull request merged**